### PR TITLE
[MNG-7566] Support Java prerequisites in plugin descriptor

### PR DESCRIPTION
**Pull request by kwin**
Depends on https://github.com/apache/maven/pull/827

### CODE REVIEW

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7566JavaPrerequisiteTest.java
Comment (kwin) [inline core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7566JavaPrerequisiteTest.java:20]: which range to take now requiring current Maven master?

Comment (michael-o) [reply core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7566JavaPrerequisiteTest.java:20]: Makes sense..

File: core-it-suite/src/test/resources/mng-7566/repo/org/apache/maven/its/mng7566/maven-mng7566-plugin/0.1/maven-mng7566-plugin-0.1.pom
Comment (michael-o) [inline core-it-suite/src/test/resources/mng-7566/repo/org/apache/maven/its/mng7566/maven-mng7566-plugin/0.1/maven-mng7566-plugin-0.1.pom:47]: Don't use stoneage deps.

File: core-it-suite/src/test/resources/mng-7566/repo/org/apache/maven/its/mng7566/maven-mng7566-plugin/0.2/maven-mng7566-plugin-0.2.pom
Comment (michael-o) [inline core-it-suite/src/test/resources/mng-7566/repo/org/apache/maven/its/mng7566/maven-mng7566-plugin/0.2/maven-mng7566-plugin-0.2.pom:47]: Same here

**Review by michael-o — CHANGES_REQUESTED** [review]

File: core-it-suite/src/test/resources/mng-7566/repo/org/apache/maven/its/mng7566/maven-mng7566-plugin/0.1/maven-mng7566-plugin-0.1.pom
Comment (kwin) [reply core-it-suite/src/test/resources/mng-7566/repo/org/apache/maven/its/mng7566/maven-mng7566-plugin/0.1/maven-mng7566-plugin-0.1.pom:47]: This is just a copy from MavenITmng4840MavenPrerequisiteTest. Why does it matter here? Which version do you suggest?

Comment (michael-o) [reply core-it-suite/src/test/resources/mng-7566/repo/org/apache/maven/its/mng7566/maven-mng7566-plugin/0.1/maven-mng7566-plugin-0.1.pom:47]: I bet that @gnodet has already updated this in his branch. The reason is to reduce variance in dependencies. Maven 2 is dead, therefore we should avoid using those artifacts even if the API hasn't changed.

Comment (kwin) [reply core-it-suite/src/test/resources/mng-7566/repo/org/apache/maven/its/mng7566/maven-mng7566-plugin/0.1/maven-mng7566-plugin-0.1.pom:47]: In master it still is 2.0: https://github.com/apache/maven-integration-testing/blob/bd7ba0adf5f847b394330cfc022ea833e022ebe5/core-it-suite/src/test/resources/mng-4840/repo/org/apache/maven/its/mng4840/maven-mng4840-plugin/0.1/maven-mng4840-plugin-0.1.pom#L51.
But I can change to 3.8.6.

Comment (michael-o) [reply core-it-suite/src/test/resources/mng-7566/repo/org/apache/maven/its/mng7566/maven-mng7566-plugin/0.1/maven-mng7566-plugin-0.1.pom]: Let's leave it like this. This can been sweeped all at once.

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7566JavaPrerequisiteTest.java
Comment (gnodet) [reply core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7566JavaPrerequisiteTest.java:20]: Next version should be `4.0.0-alpha-3`, but how are snapshots taken into account ?

File: core-it-suite/src/test/resources/mng-7566/repo/org/apache/maven/its/mng7566/maven-mng7566-plugin/0.1/maven-mng7566-plugin-0.1.pom
Comment (gnodet) [reply core-it-suite/src/test/resources/mng-7566/repo/org/apache/maven/its/mng7566/maven-mng7566-plugin/0.1/maven-mng7566-plugin-0.1.pom]: https://github.com/apache/maven-integration-testing/blob/582dd0199262595fb647eb34c05d60aa4ad86941/core-it-suite/src/test/resources/mng-4840/repo/org/apache/maven/its/mng4840/maven-mng4840-plugin/0.1/maven-mng4840-plugin-0.1.pom#L51

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7566JavaPrerequisiteTest.java
Comment (kwin) [reply core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7566JavaPrerequisiteTest.java:20]: Certain patterns are stripped before comparing with the given version range: https://github.com/apache/maven-integration-testing/blob/bd7ba0adf5f847b394330cfc022ea833e022ebe5/core-it-support/maven-it-helper/src/main/java/org/apache/maven/it/AbstractMavenIntegrationTestCase.java#L189

SNAPSHOT is one of them.

**Comment by gnodet** [comment]
@kwin I think this PR should have been merged along with https://github.com/apache/maven/pull/827

**Comment by kwin** [comment]
@gnodet I was waiting for approval but merged it now!

**Pull request merged**
